src/services/RedisService.ts

```diff
@@ -0,0 +1,136 @@
+import { inject, Container } from 'inversify';
+import { provide } from 'inversify-binding-decorators';
+import { TYPES } from '../InjectableTypes';
+import { IMap } from '../utils/Types';
+import { promisify } from 'util';
+import { createClient, RedisClient } from 'redis';
+
+@provide(TYPES.RedisService)
+export class RedisService {
+
+    private healthObject: any = {
+        n: 0,
+        lastTimeVisited: new Date().getTime().toString()
+    };
+
+    constructor(@inject(TYPES.RedisClient) private readonly redisClient: RedisClient) {
+        this.redisClient.on('connect', () => this.checkHealth);
+    }
+
+    public async checkHealth(): Promise<any> {
+        return this.getObject('healthCheck').then(obj => {
+            if (obj) {
+                this.healthObject = obj;
+                this.healthObject.n = Number(this.healthObject.n) + 1;
+                this.healthObject.lastTimeVisited = new Date().getTime().toString();
+                this.setObject('healthCheck', this.healthObject);
+            } else {
+                this.setObject('healthCheck', this.healthObject);
```

Comment:
So if there is no connection to redis, then we will still return the lastTimeVisited?

Reply:
![image](https://user-images.githubusercontent.com/54070318/73744514-d8477300-4748-11ea-8588-16e7448a23cf.png)

Reply:
Yes

Reply:
Could we format the date into something more readable?

Reply:
As far as service is concerned IMO it should just say healthy / unhealthy, nothing more.